### ISSUE TITLE
TRVs loosing connection in Lovelace when switchung AUTO to HEAT mode

### ISSUE THREAD

**Issue by Merc3nary** [SukramJ/hahomematic#388]
DONT'T DELETE THIS.
Please answer all questions as good as possible.

Make sure you have read the documentation at https://github.com/danielperna84/custom_homematic#custom_homematic.

Make sure you are running the latest version of hahomematic or it's custom_component (and Home Assistant) before reporting an issue.

**custom_component/hahomematic version (if applicable):**
1.0.3

**Home Assistant version (if applicable):**
2022.3.8

**CCU version:**
- Model (Raspi/4):
- OS (RaspberryMatic on Homeassistant OS):
- Version: 5.10.103-v8 
(hope this is the info you needed)

**Problem-relevant configuration:**

Do you use tls? no
Do you use callback? no
Do you use username and password? yes
Which interfaces do you use?
the bug refers to Lovelace if this is what you want to know here

**Describe the bug**

I am using 5 HmIP-eTRV-2 and have added them with climate cards to Lovelace.
In general the interaction between the Lovelace card and the hardware works quite well, sometimes with a bit of delay but in general the TRV reacts to changes in the manual temperature and the boost. Likewise turning the wheel on the TRV changes the settings in the Lovelace card. 
However, when switching from manual to auto (or the other way around) the Lovelace card looses connection, displays N/A and jumps to the "NONE" preset. At that point it does not react anymore.
Turning the wheel on the TRV will reactivate the Lovelace card but it is more likely to get more N/A situations afterwards.
The biggest issue seems to be the change from MANUAL to AUTO.  

**To Reproduce**
1. Go to 'a Lovelace card of the TRV'
2. Click on 'AUTO if it is in HEAT, or HEAT when it is in AUTO'
3. The Lovelacecard will go into N/A display and the TRV will not react anymore.

Before
![image](https://user-images.githubusercontent.com/49135121/161618014-a27ae344-8e52-48b6-9d10-c6c6e6ce3d72.png)

after clicking on HEAT
![image](https://user-images.githubusercontent.com/49135121/161618221-6e66a54d-0f80-4dd0-a14e-f9750f7e19d8.png)

**Screenshots (if applicable)**
- CCU firewall
- integrations view
- device view
- enity view

**Relevant HA log entries (full log)**
```
which logs do you want to have?

```

**Diagnostic Information (available on every device)**
```


```

**Expected behavior**
No loosing of connection

**Screenshots**

**Additional context**

**Comment by SukramJ** [comment]
Which lovelace card is this? Please try the default one.
I cannot reproduce this with my HmIP-eTRV-2.

> which logs do you want to have?

The default HA /core log

**Comment by Merc3nary** [comment]
OK I tried it with the standard climate card and it behaves the same way.

The card I was using is a custom card called "simple Thermostat"

**Comment by SukramJ** [comment]
What about logs?
Please check the device display **AND** the ccu web ui if there are any error for your devices.

**Comment by Merc3nary** [comment]
Will have to post the logs later.
The TRV itself does not loose connection and as soon as I turn or press the wheel the card comes back to life.

**Comment by SukramJ** [comment]
> Please check the device display AND the ccu web ui if there are any error for your devices.

Do you really think i am asking this for no reason?

**Comment by SukramJ** [comment]
Hi @noxhirsch, @eikowagenknecht, @Lemocin 
you also you HmIP-eTRV-?. Can you reproduce this behaviour?

Thanks in advance.

**Comment by Lemocin** [comment]
> you also you HmIP-eTRV-?. Can you reproduce this behaviour?

Nope, working fine.
But when i switch to Auto, i get another Error and Profile gets empty:

![image](https://user-images.githubusercontent.com/4162496/161719792-d12a8efa-27d4-4341-a9e3-313b8bc2a1e2.png)

When i open the card again it shows fine, so no problem for me.

**Comment by SukramJ** [comment]
@Lemocin i have seen this message too. I think it has something to do with the frontend validation.

**Comment by SukramJ** [comment]
> **CCU version:**
> 
> * Model (Raspi/4):
> * OS (RaspberryMatic on Homeassistant OS):
> * Version: 5.10.103-v8
>   (hope this is the info you needed)

@Merc3nary 
Which RF-Module do you use and how is it connected?

**Comment by noxhirsch** [comment]
> Can you reproduce this behaviour?

@SukramJ: I also can not reproduce this behaviour (but I get the error message regarding `preset_mode` as well)

**Comment by Merc3nary** [comment]
Ok, I am back from work now and have a bit more time and the possibility to answer in more detail.

My RF-Module is the one from ELV: ELV Bausatz Homematic IP Funkmodul für Raspberry Pi HM-MOD-RPI-PCB
[https://de.elv.com/elv-homematic-komplettbausatz-funkmodul-fuer-raspberry-pi-hm-mod-rpi-pcb-fuer-smart-home-hausautomation-142141?fs=4095418356](url)
At the moment the Pi where this is installed has no housing that should have an impact on the antenna.

Here is how the connection to the 5 TRVs looks in RaspberryMatic

![grafik](https://user-images.githubusercontent.com/49135121/161791646-16bd9531-0cfb-4f5f-b3bc-920ac2002aa9.png)


As you can see one is UNREACH at the moment and that happend after I clicked on the AUTO button in Lovelace.

Here is what I did:

1. Go to Lovelace, all TRV showed the temperature and looked good
2. I clicked on the AUTO button of the living room TRV
3. as a result the Lovelace card switched to NA
4. in RaspberryMatic the TRV turned to UNREAD and I get a yellow service message
![grafik](https://user-images.githubusercontent.com/49135121/161792249-da31863e-a0a1-412f-8f76-6144636743af.png)
5. On the TRV the little antenna symbol in the lower right starts flashing
6. the moment I clicked on it and it went to N/A the following log entry appeared in HA:

Logger: hahomematic.client
Source: /usr/local/lib/python3.9/site-packages/hahomematic/client.py:414
First occurred: 17:31:12 (1 occurrences)
Last logged: 17:31:12
set_value failed with ProxyException [(<Fault -1: 'Generic error (TIMEOUT)'>,)]: 000A1D89A65307:1, SET_POINT_TEMPERATURE, 10.0

When I turn the dial on the TRV the Lovelace card gets updated again and the service message in RaspberryMatic disappears.

I hope this is helpful and includes the necessary information. If not please let me know - it certainly does not happen with bad intention but the tone a bit earlier put me off.

I have tested it with the standard climate card as well and the same thing happens.
I also do NOT get the preset_mode error message.

Cheers,
Merc

**Comment by SukramJ** [comment]
You may have noticed that other users cannot confirm the problem

I think your problem is more related to your setup.
Check the [homematic forum](https://homematic-forum.de/forum/index.php) for corresponding reports about problems with RPi4 and directly attached RF-Modules.
Good start is [this](https://homematic-forum.de/forum/viewtopic.php?f=31&t=65293) from @baxxy13.
If there is a (temporary) communication problem between ccu and device we get an unreach notification from ccu. That's why it's greyed out. Next request might be OK again to send to the device.

**Comment by Merc3nary** [comment]
@SukramJ,
thanks for the link I will check it out.
Good for the other users but I can just report my experience.
Is there a better way with respect to the RF-modules? I do not have to stay with this one if other setups are less error prone.

Cheers,
Merc

**Comment by SukramJ** [comment]
Reporting is always good.

Just read the linked article from baxxy, if german is no problem.
You should have a minimum distance of 1m between RPi4 and RF-Modul.

Good option is the [USB-Stick](https://de.elv.com/elv-homematic-ip-arr-bausatz-rf-usb-stick-fuer-alternative-steuerungsplattformen-hmip-rfusb-fuer-smart-home-hausautomation-152306?fs=2222724441&c=691) with RaspberryMatic.

**Comment by Merc3nary** [comment]
There is one more thing I would lie to mention @SukramJ,
this error does only appear when switching between Heat and Auto. I can manually dial up and down the temperature as much as I want, I can turn the TRV off from the Lovelace card and I can trigger the boost. Also the CCU manages to trigger the different temperatures I have saved in the programs and nothing bad happens.
To me it is a bit strange that this should be a connectivity problem.

**Comment by SukramJ** [comment]
RF-Problems are not consistent over time an also the corresponding device commands maybe implemented different.
As long as I and  others cannot reproduce this issue i don't believe in a problem in this integration.

I'll close this issue for now. Please feel free to reopen this issue if the problem exists **after** you fixed your setup.

**Comment by Merc3nary** [comment]
@SukramJ,
I reattached the ferrit ring that comes with the RF-Module closer to the pi and made it a bit tighter and that solved the problem. Next I would have tried to use jumper cables and detach the module but apparently this was already enough. I switched the auto and heat options back and forth many times and across all the TRVs and they did not loose connection. Also I still do not get the other error message.

Thanks for the advice and for the nice custom integration.

Merc
